searchlib/src/vespa/searchlib/docstore/visitcache.cpp

```diff
@@ -306,19 +308,21 @@ VisitCache::Cache::onRemove(const K & key) {
 vespalib::MemoryUsage
 VisitCache::Cache::getStaticMemoryUsage() const {
     vespalib::MemoryUsage usage = Parent::getStaticMemoryUsage();
-    auto cacheGuard = getGuard();
     size_t baseSelf = sizeof(_lid2Id) + sizeof(_id2KeySet);
     usage.incAllocatedBytes(baseSelf);
-    usage.incAllocatedBytes(_lid2Id.capacity() * sizeof(LidUniqueKeySetId::value_type));
-    usage.incAllocatedBytes(_id2KeySet.capacity() * sizeof(IdKeySetMap::value_type));
     usage.incUsedBytes(baseSelf);
-    usage.incUsedBytes(_lid2Id.size() * sizeof(LidUniqueKeySetId::value_type));
-    usage.incUsedBytes(_id2KeySet.size() * sizeof(IdKeySetMap::value_type));
+    return usage;
+}
+
+CacheStats
+VisitCache::Cache::get_stats() const {
+    CacheStats stats = Parent::get_stats();
+    stats.memory_used += _lid2Id.capacity() * sizeof(LidUniqueKeySetId::value_type);
```

Comment:
We must take the guard here, as done in getStaticMemoryUsage previously.

Reply:
Fixed